### DISCUSSION TITLE
Larger contour plot is not working.

### DISCUSSION THREAD

**Discussion by edromzans** [gnudatalanguage/gdl#1151] [Q&A]
Hi 
when I want make a contour plot in GDL, it hangs.
A simple contour, for example:
Z = DIST(100)
CONTOUR, Z, NLEVELS=10, /FOLLOW, TITLE='Simple Contour Plot'
It is working!
But, if
Z = DIST(500) or more, It hangs.

Are you having the same problem? Are there some solution for this larger contour plots?

**Comment by GillesDuvert** [comment]
We already know about #851 , deemed a [plplot](https://sourceforge.net/p/plplot/bugs/) bug, and, unless a patch is provided by the plplot guys, not going to be cured alas. But the #851 problem arises when the contours are very complicated, which is not the case with a smooth DIST(). #851 is also particular: adding the 'FILL' option there is no more crash (the plplot function called is different)? (a test you can make).

Now, with my venerable laptop z=dist(500) gives an instant plot. Do you have a peculiar setup? memory size, type of OS (mac, linux, windows), version of GDL.
Also, with unix/linux, the allowed size of the stack ( returned by `ulimit -s` is the limit to the number of reentrant functions. Mine value is 8192)

**Reply by edromzans** [reply]
Hi, thanks for tell me about #851.
I am using GDL Version 1.0.0, compiled from git tag 1.0.0, in debian 11. My computer have 8G of memory. 
My output to 
$ulimit -s
is
8192
With /fill I can plot larger z, =dist(1500) for example. But my real necessity is plot more complicate contour over maps and in this cases I am have been hangs same using /fill.

"Complicate" contours spend a lot of time until stop (hangs) and I have to force close GDL. I hope that the plplot guys can found some solution to #851.

I must test this contours with other devices to check.

**Comment by GillesDuvert** [comment]
Of course if all contours were as boring as DIST() nobody would want to see them :smile: 
Would you try CONTOUR,/FILL on your "real necessity" and tell us if they work?
If so a patch could be (TBC) to use the '/fill' engine but without actually filling the contours.
